execution-engine/engine-tests/src/test/system_contracts/client_api_proxy.rs

```diff
@@ -68,6 +74,74 @@ fn should_invoke_successful_transfer_to_account() {
     assert_eq!(balance, U512::from(TRANSFER_AMOUNT));
 }
 
+#[ignore]
+#[test]
+fn should_invoke_successful_standard_payment() {
+    // run genesis
+    let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&*DEFAULT_GENESIS_CONFIG).commit();
+
+    let client_api_proxy_hash = get_client_api_proxy_hash(&builder);
+
+    // transfer 1 from DEFAULT_ACCOUNT to ACCOUNT_1
+    let payment_amount = 10_000_000;
```

Comment:
Do you know why the problem occurs when payment_amount is 100_000_000? I tried to change the problem, but I do not know why.

Reply:
Oh, there was a type mismatch. fixed in b4117dd.

Reply:
Additionally, do you know why only payment amount is u32?
 I don't know why only the payment was handled by casperlabs like this.

Reply:
I guess there is no special reason for that.
In contract-api, the arguments related to amount are unified with U512, but just for convenience, they use u32, u64 instead of U512 in contract.(By doing this, we can have less verbose json). It is actually possible to pass them as U512.